### DISCUSSION TITLE
Photos - Resync again after deletion

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#3242] [Enhancements]
[Moved from Discord]

### Description

Steps to reproduce:
- sync all your iOS photos with Ente Photos
- delete photos from Ente Photos on macOS, empty trash and kill app
- relaunch iOS app, the deleted photos start synching again
- reopen macOS and notice that all the deleted photos are here again...

This is not the first it happens. I took a lot of time selecting and deleting A LOT of pictures on macOS. Now I have to start again and am very afraid to lose time again :/ 

### What product are you using?

Ente Photos

### What platform are you using?

Mobile - iOS, Desktop - macOS

**Comment by vishnukvmd** [comment]
## Proposed workaround

- sync photos on iOS and macOS Ente Photos apps
- close both apps
- open macOS Ente Photos application
- delete photos from macOS Ente Photos application  without emptying trash
- open iOS Ente Photos application and let it sync trash history
- once trash history is synched, empty trash from iOS

**Comment by paulhovey** [comment]
I just noticed something that might or might not be related.  I'd deleted a bunch of photos from the desktop app (on Windows) and let the trash sync (did not empty the trash) with the iPhone app so it knew not to re-upload those photos.  I think the 30 day trash expiry just occurred, because now the iOS app is wanting to re-upload all the photos that previously had been deleted from the desktop app.

**Comment by ua741** [comment]
>  because now the iOS app is wanting to re-upload all the photos that previously had been deleted from the desktop app.

Once the trash entry is synced, the app should not automatically reupload those entries that were previous deleted. When you open the device folder, you should still see the message that some of the items are ignored for delete because they were previously deleted.
If this is not happening, then it looks like a regression.

**Comment by paulhovey** [comment]
This behavior also occurs when getting a new iPhone.  Since I had the "Recents" folder backed up in order to capture all still/live photos and videos, there were a lot of screenshots that were also uploaded that were unnecessary.  Since all photos were transferred to the new phone at time of purchase, when ente is re-installed on the new iPhone and "Recents" is selected to be backed up all the previously ignored/deleted screenshots and photos will be backed up again.  This is very annoying behavior, because now my two options are either leaving ente as-is and having hundreds of old screenshots saved to ente, or letting ente re-upload years of old photos, then spending the time to clear them out again.

Suggestion: having some saved list of images that are deleted that is synced between the desktop and mobile apps would solve all these problems, instead of using the trash and hoping people don't immediately empty the trash from desktop or get a new iPhone and have photos previously deleted.

**Comment by ua741** [comment]
Similar/Related requests:
https://github.com/ente-io/ente/discussions/3425
https://github.com/ente-io/ente/discussions/3546
https://github.com/ente-io/ente/discussions/900
